app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -325,6 +319,25 @@ class ThreadFragment : Fragment() {
         threadAdapter.stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
     }
 
+    private fun onAttachmentClicked(attachable: Attachable) {
+        if (attachable is Attachment) {
+            trackAttachmentActionsEvent(ACTION_OPEN_NAME)
+            attachable.openAttachment(
+                context = requireContext(),
+                navigateToDownloadProgressDialog = { attachment, attachmentIntentType ->
+                    navigateToDownloadProgressDialog(
+                        attachment,
+                        attachmentIntentType,
+                        ThreadFragment::class.java.name,
+                    )
+                },
+                snackbarManager = snackbarManager,
+            )
+        } else if (attachable is SwissTransferFile) {
+            downloadSwissTransferFile(swissTransferFile = attachable)
```

Comment:
This lacks a Matomo tracker, talks about it with @LunarX and @valentinperignon to be iso with iOS, and they'll show you the different steps to do when adding a matomo tracker, like completing the documentation